projects/firebaseui-angular-library/src/lib/firebaseui-angular-library.service.ts

```diff
@@ -1,20 +1,234 @@
-import {Inject, Injectable, NgZone, Optional} from '@angular/core';
+import { EventEmitter, Inject, Injectable, NgZone, Optional } from '@angular/core';
+import { FIREBASE_APP_NAME, FIREBASE_OPTIONS, FirebaseApp, FirebaseAppConfig, FirebaseOptions, ɵfirebaseAppFactory } from '@angular/fire';
 import * as firebaseui from 'firebaseui';
-import {FIREBASE_APP_NAME, FIREBASE_OPTIONS, FirebaseApp, FirebaseAppConfig, FirebaseOptions, ɵfirebaseAppFactory} from '@angular/fire';
+import { DynamicLoaderService, Resource } from './dynamic-loader.service';
+import { ExtendedFirebaseUIAuthConfig, FirebaseUILanguages, FirebaseUISignInFailure, FirebaseUISignInSuccessWithAuthResult, } from './firebaseui-angular-library.helper';
+
+declare const global: any;
+
+export const DEFAULT_FIREBASE_UI_AUTH_CONTAINER = "#firebaseui-auth-container";
 
 @Injectable()
 export class FirebaseuiAngularLibraryService {
-  public firebaseUiInstance: firebaseui.auth.AuthUI;
 
-  constructor(@Inject(FIREBASE_OPTIONS) options: FirebaseOptions,
-              @Optional() @Inject(FIREBASE_APP_NAME) nameOrConfig: string | FirebaseAppConfig | null | undefined,
-              zone: NgZone) {
+  private static readonly COMPUTED_CALLBACKS = "COMPUTED_CALLBACKS";
+  private static readonly FIREBASEUI_CDN_VERSION = "4.5.0";
```

Comment:
This version can't be hardcoded.
1. I would need to update this string on every release.
2. The firebaseui library is as such independent from this wrapper plugin, and as such everyone who uses this wrapper could use a different verison.